Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm

```diff
@@ -47,20 +50,56 @@
 }
 
 #if HAVE(WEB_TRANSPORT)
-static RetainPtr<nw_parameters_t> createParameters(WebCore::SecurityOriginData&& origin)
+
+static void didReceiveServerTrustChallenge(Ref<NetworkConnectionToWebProcess>&& connectionToWebProcess, URL&& url, WebKit::WebPageProxyIdentifier&& pageID, WebCore::SecurityOriginData&& topOrigin, sec_trust_t trust, sec_protocol_verify_complete_t completion)
+{
+    const char * host = url.host().utf8().data();
+    uint16_t port = url.port() ? *url.port() : 443;
+    RetainPtr secTrust = adoptCF(sec_trust_copy_ref(trust));
+    RetainPtr protectionSpace = adoptNS([[NSURLProtectionSpace alloc] initWithHost:[NSString stringWithUTF8String:host] port:port protocol:NSURLProtectionSpaceHTTPS realm:nil authenticationMethod:NSURLAuthenticationMethodServerTrust]);
```

Comment:
Yeah, this host use is a UAF.  Let's use initWithHost:url.host().createNSString().get() instead.